v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/schemautils/PubSubUtils.java

```diff
@@ -0,0 +1,396 @@
+import org.apache.avro.io.EncoderFactory;
+import org.apache.commons.lang3.Validate;
+import org.json.JSONObject;
+
+/**
+ * {@link PubSubUtils} provides utils for processing PubSub schema and generating PubSub messages.
+ */
+public class PubSubUtils implements Serializable {
+
+  private static final EnumMap<PubSubFields, PubSubValueFormatter> FORMATTERS =
+      new EnumMap<>(PubSubFields.class);
+
+  static {
+    FORMATTERS.put(
+        PubSubFields.ROW_KEY_STRING,
+        (pb, chg) -> {
+          String rowkeyEncoded = chg.getString(PubSubFields.ROW_KEY_BYTES.name());
+          return pb.convertBase64ToString(rowkeyEncoded);
+        });
+    FORMATTERS.put(
+        PubSubFields.ROW_KEY_STRING_BASE64, (pb, chg) -> PubSubFields.ROW_KEY_STRING_BASE64);
+    FORMATTERS.put(
+        PubSubFields.ROW_KEY_BYTES,
+        (pb, chg) -> {
+          String rowkeyEncoded = chg.getString(PubSubFields.ROW_KEY_BYTES.name());
+          return pb.convertBase64ToBytes(rowkeyEncoded);
+        });
+    FORMATTERS.put(PubSubFields.MOD_TYPE, (pb, chg) -> chg.getString(PubSubFields.MOD_TYPE.name()));
+    FORMATTERS.put(
+        PubSubFields.COMMIT_TIMESTAMP,
+        (pb, chg) -> chg.getLong(PubSubFields.COMMIT_TIMESTAMP.name()));
+    FORMATTERS.put(
+        PubSubFields.COLUMN_FAMILY, (pb, chg) -> chg.getString(PubSubFields.COLUMN_FAMILY.name()));
+    FORMATTERS.put(
+        PubSubFields.COLUMN,
+        (pb, chg) -> {
+          if (!chg.has(PubSubFields.COLUMN.name())) {
+            return null;
+          }
+          String qualifierEncoded = chg.getString(PubSubFields.COLUMN.name());
+          return pb.convertBase64ToBytes(qualifierEncoded);
+        });
+    FORMATTERS.put(
+        (PubSubFields.COLUMN_STRING_BASE64), (pb, chg) -> PubSubFields.COLUMN_STRING_BASE64);
+    FORMATTERS.put(
+        PubSubFields.TIMESTAMP,
+        (pb, chg) -> {
+          if (!chg.has(PubSubFields.TIMESTAMP.name())) {
+            return null;
+          }
+          return chg.getString(PubSubFields.TIMESTAMP.name());
+        });
+    FORMATTERS.put(
+        PubSubFields.TIMESTAMP_NUM,
+        (pb, chg) -> {
+          if (!chg.has(PubSubFields.TIMESTAMP_NUM.name())) {
+            return null;
+          }
+          return chg.getLong(PubSubFields.TIMESTAMP_NUM.name());
+        });
+    FORMATTERS.put(
+        PubSubFields.VALUE_STRING,
+        (pb, chg) -> {
+          if (!chg.has(PubSubFields.VALUE_BYTES.name())) {
+            return null;
+          }
+
+          String valueEncoded = chg.getString(PubSubFields.VALUE_BYTES.name());
+          return pb.convertBase64ToString(valueEncoded);
+        });
+    FORMATTERS.put(
+        PubSubFields.VALUE_BYTES,
+        (pb, chg) -> {
+          if (!chg.has(PubSubFields.VALUE_BYTES.name())) {
+            return null;
+          }
+
+          String valueEncoded = chg.getString(PubSubFields.VALUE_BYTES.name());
+          return pb.convertBase64ToBytes(valueEncoded);
+        });
```

Comment:
Formatter for VALUE_STRING and VALUE_BYTES have identical implementations. I think we want to use VALUE_STRING for JSON and we probably should add VALUE_STRING_BASE64 as well so we have uniformity with how we implement rowkeys and column qualifiers.